### PR TITLE
feat: Add sourceDescription to ns-workflows-1

### PR DESCRIPTION
**Pull request by frankkilcommins**
Refs: #3392

Add SourceDescription object to apidom-ns-workflows-1

### CODE REVIEW

File: packages/apidom-ns-workflows-1/src/traversal/visitor.ts [hunk 1]
```diff
@@ -15,5 +15,6 @@ export const getNodeType = <T extends Element>(element: T): string | undefined =
 export const keyMap = {
   WorkflowsSpecification1Element: ['content'],
   InfoElement: ['content'],
+  isSourceDescriptionElement: ['content'],
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/traversal/visitor.ts:18]:
```suggestion
  SourceDescriptionElement: ['content'],
```

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/sourceDescription/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import stampit from 'stampit';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/sourceDescription/index.ts:1]: Please change the name of the directory from `workflows-1/sourceDescription` to `workflows-1/source-description`. By convention all directories are `dash-cased`.

File: packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts [hunk 3]
```diff
@@ -26,6 +26,14 @@ class WorkflowsSpecification1 extends ObjectElement {
   set info(info: InfoElement | undefined) {
     this.set('info', info);
   }
+
+  get sourceDescriptions(): ArrayElement | undefined {
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts:30]: We ca have more specific `NCE` for this list of source description, let's use it.

```suggestion
  get sourceDescriptions(): SourceDescriptionsElement | undefined) {
```

File: packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts [hunk 4]
```diff
@@ -26,6 +26,14 @@ class WorkflowsSpecification1 extends ObjectElement {
...
+    return this.get('sourceDescriptions');
+  }
+
+  set sourceDescriptions(sourceDescriptions: ArrayElement | undefined) {
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts:34]: We ca have more specific `NCE` for this list of source description, let's use it.

```suggestion
  get sourceDescriptions(): SourceDescriptionsElement | undefined) {
```

File: packages/apidom-ns-workflows-1/src/refractor/specification.ts [hunk 5]
```diff
@@ -37,6 +42,14 @@ const specification = {
             version: InfoVersionVisitor,
           },
         },
+        SourceDescription: {
+          $visitor: SourceDescriptionVisitor,
+          fixedFields: {
+            name: FallbackVisitor,
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/specification.ts:48]:
```suggestion
            name: { $ref: '#/visitors/value' },
```

File: packages/apidom-ns-workflows-1/src/refractor/predicates.ts [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
-import { MemberElement, isStringElement, toValue } from '@swagger-api/apidom-core';
+import { MemberElement, isStringElement, isObjectElement, toValue } from '@swagger-api/apidom-core';
 
 // eslint-disable-next-line import/prefer-default-export
 export const isWorkflowsSpecificationExtension = (element: MemberElement): boolean => {
   // @ts-ignore
   return isStringElement(element.key) && startsWith('x-', toValue(element.key));
 };
+
+export const isSourceDescriptionLikeElement = isObjectElement;
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/predicates.ts:10]: We don't need this predicate.
```suggestion
```

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/SourceDescriptionsVisitor.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import FallbackVisitor from '../FallbackVisitor';
+import { isSourceDescriptionLikeElement } from '../../predicates';
+
+const SourceDescriptionsVisitor = stampit(SpecificationVisitor, FallbackVisitor, {
+  init() {
+    this.element = new SourceDescriptionsElement();
+  },
+  methods: {
+    ArrayElement(arrayElement: ArrayElement) {
+      arrayElement.forEach((item: Element) => {
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/SourceDescriptionsVisitor.ts:15]: We don't need to assert the shape here. The look can look like this:

```ts
      arrayElement.forEach((item: Element): void => {
        const specPath = ['document', 'objects', 'SourceDescription'];
        const element = this.toRefractedElement(specPath, item);

        this.element.push(element);
      });
```

The reason is that the [SourceDescriptionVisitor](https://github.com/swagger-api/apidom/pull/3403/files#diff-f3530852a6bd15c65b79b08c2a0a74408bad87c3e8ac52c8acd48a65dbdc9538R8) contains `FallbackVisitor` in it's prototype chain and will either refract to semantic `SourceDescription` or generic value.

File: packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/sourceDescription/TypeVisitor.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import stampit from 'stampit';
+import { StringElement, BREAK, cloneDeep } from '@swagger-api/apidom-core';
+
+import FallbackVisitor from '../../FallbackVisitor';
+
+const TypeVisitor = stampit(FallbackVisitor, {
+  methods: {
+    StringElement(stringElement: StringElement) {
+      this.element = cloneDeep(stringElement);
+      this.element.classes.push('sourceDescription-type');
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/sourceDescription/TypeVisitor.ts:10]: If you'll use classes here, the convention is to use `dash-case`: 
```suggestion
      this.element.classes.push('source-description-type');
```

**Review by char0n — CHANGES_REQUESTED** [review]
Looking good! Some changes needed before the merge

File: packages/apidom-ns-workflows-1/src/index.ts [hunk 9]
```diff
@@ -43,4 +45,5 @@ export {
   WorkflowsSpecification1Element,
   WorkflowsSpecElement,
   InfoElement,
+  SourceDescriptionElement,
 } from './refractor/registration';
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/index.ts:49]: `SourceDescriptionsElement` NCE is not exported from here

File: packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts (hunk 3 above)
Comment (char0n) [reply packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts:30]: Wasn't it possible to use `SourceDescriptionsElement` here?

**Review by char0n — APPROVED** [review]
LGTM. Left [1 comment](https://github.com/swagger-api/apidom/pull/3403#discussion_r1397111759) to possibly look at before the merge.

File: packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts (hunk 3 above)
Comment (char0n) [reply packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts:30]: All right, let's disregard it. Other namespaces are not doing this.

File: packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts (hunk 4 above)
Comment (char0n) [reply packages/apidom-ns-workflows-1/src/elements/WorkflowsSpecification1.ts:34]: All right, let's disregard it. Other namespaces are not doing this.

File: packages/apidom-ns-workflows-1/test/predicates.ts [hunk 10]
```diff
@@ -5,9 +5,11 @@ import {
   isWorkflowsSpecification1Element,
   isWorkflowsSpecElement,
   isInfoElement,
+  isSourceDescriptionElement,
   WorkflowsSpecification1Element,
   WorkflowsSpecElement,
   InfoElement,
```
[the remark this answers is not in the document]
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/test/predicates.ts:12]: Added the missing test. Thanks for spotting.
FYI - similar tests are missing in OAS3 and 3.1 namespaces for the `isServersElement` predicate

**Review by char0n — APPROVED** [review]
Thank you for the test @frankkilcommins!

**Pull request merged**